src/demo/ui.jsx

```diff
@@ -543,8 +546,26 @@ class Words extends React.Component {
   constructor(props) {
     super(props);
 
-    // Randomize word choices and set in state.
-    const choices = _.shuffle(wordSet[getState().appMode].choices);
+    // Randomize word choices in each set, merge the sets, and set as state.
+    const appMode = getState().appMode;
+    const appModeWordSet = wordSet[appMode].choices;
+    let choices = [];
+    let maxSize = 0;
+    // Each subset represents a different column, so merge the sets
+    for (var i = 0; i < appModeWordSet.length; ++i) {
+      appModeWordSet[i] = _.shuffle(appModeWordSet[i]);
+      if (appModeWordSet[i].length > maxSize) {
+        maxSize = appModeWordSet[i].length;
+      }
+    }
+    for (i = 0; i < maxSize; ++i) {
```

Comment:
What's the reason for this specific nested loop - does `choices` need to be in a certain order?

Reply:
Yeah. The colors all need to be in the left column and the shapes in the right.

Reply:
But the local variable `choices` that you set in state is just a flat list which is mixed between the two, right?

Reply:
In order to get it to show up correctly it needs to be separated into the two columns. I was trying to figure out the CSS to do this "better" but couldn't maintain the button styles correctly.

Reply:
the order only matters for display reasons, so i think it's okay that the end result is a flat list

Reply:
👍 bethany explained it to me on slack.